lib/xapi/file_bundle.rb

```diff
@@ -0,0 +1,28 @@
+require 'zip'
+module Xapi
+  # Module to return all the files in given path and zip them
+  class FileBundle
+    attr_reader :dir, :ignore_patterns
+    def initialize(dir, ignore_patterns=[])
+      @dir = dir
+      @ignore_patterns = ignore_patterns
+    end
+
+    def create_zip
+      Zip::OutputStream.write_buffer do |io|
+        list_paths.each do |path|
+          io.put_next_entry(path.relative_path_from(dir))
+          io.print IO.read(path)
+        end
+        yield io if block_given?
+      end
+    end
+
+    def list_paths
```

Comment:
Same here. This is a command to do something. But the outside caller shouldn't be allowed to micromanage how and when someone does something, when all it wants is some information back.

There are situations when the correct name might be _do X_, but in those cases I would expect the method to have a side effect, not a return value.